nvi-commons/src/main/java/no/sikt/nva/nvi/common/service/NviService.java

```diff
@@ -138,10 +140,23 @@ private boolean isNewCandidate(DbCandidate dbCandidate) {
     @JacocoGenerated // bug in jacoco report that is unable to exhaust the switch. Should be fixed in version 0.8.11
     private DbApprovalStatus toUpdatedApprovalStatus(DbApprovalStatus oldApprovalStatus,
                                                      DbApprovalStatus newApprovalStatus) {
-        return switch (newApprovalStatus.status()) {
-            case APPROVED, REJECTED -> updateStatus(oldApprovalStatus, newApprovalStatus);
-            case PENDING -> resetStatus(oldApprovalStatus);
-        };
+        if (oldApprovalStatus.status().equals(newApprovalStatus.status())) {
+            return updateAssignee(oldApprovalStatus, newApprovalStatus);
+        } else {
+            return switch (newApprovalStatus.status()) {
+                case APPROVED, REJECTED -> updateStatus(oldApprovalStatus, newApprovalStatus);
+                case PENDING -> resetStatus(oldApprovalStatus);
+            };
+        }
+    }
```

Comment:
In my opinion this object should be call Approval and not ApprovalStatus
And then we can check if update is status, assignee or note and refactor it to do right things